### ISSUE TITLE
Remove SEO Functionality

### ISSUE THREAD

**Issue by rclations** [INN/largo#1437]
Get the theme out of the way to other plugins can take over SEO functionality

**Comment by cartonic** [comment]
Trying to post something similar saw that is already here. I want to use a seo plugin with Largo but apparently in the <title> meta tag, my site title appears twice. Outputted by the plugin and by the theme. Is it something I can override to fix this?

**Comment by benlk** [comment]
@cartonic What you describe is this issue: https://github.com/INN/largo/issues/1470

The temporary fix for this is copying Largo's `header.php` into your child theme and removing the `<title> ... </title>` section: https://github.com/INN/umbrella-rivard-report/pull/14/files

**Comment by cartonic** [comment]
benlk Thank you for the help. I'll do that for now.

**Comment by benlk** [comment]
Would we want to remove the SEO functionality in `inc/opengraph.php` entirely, or continue to include it by default but have a checkbox/toggle that lets people disable it if they don't want it?

**Comment by cartonic** [comment]
A checkbox would be awesome since some people could rely on this only. Since I use YOAST that already has the functionality I would delete that content for now. 

    On Friday, September 15, 2017 9:46 PM, Ben Keith <notifications@github.com> wrote:
 

 Would we want to remove the SEO functionality in inc/opengraph.php entirely, or continue to include it by default but have a checkbox/toggle that lets people disable it if they don't want it?—

**Comment by aschweigert** [comment]
I would still argue that having a sane set of defaults that don't necessarily rely on a plugin is a good idea, particularly for very small sites. The option to modify the defaults, filter the output of various functions, etc. would definitely be preferred though to avoid potential collisions like the one described here.

**Comment by cartonic** [comment]
Yes, exactly. If possible a toggle in the admin panel would be the best solution. Who wants to use a dedicated plugin to disable that feature.

    On Saturday, September 16, 2017 12:28 AM, Adam Schweigert <notifications@github.com> wrote:
 

 I would still argue that having a sane set of defaults that don't necessarily rely on a plugin is a good idea, particularly for very small sites. The option to modify the defaults, filter the output of various functions, etc. would definitely be preferred though to avoid potential collisions like the one described here.—

**Comment by rclations** [comment]
a good start would be to detect Yoast and automatically hide those settings if it's activated.

**Comment by benlk** [comment]
Full fix in Largo:

- [ ] checkbox, disabled by default, to suppress Largo's own open graph tags output
- [ ] if Yoast is active, a notice in that settings area that recommends checking it to disable Largo's own meta graph stuff.
- [ ] if checkbox is not checked, do the add_action. 
- [ ] maybe some sort of output-buffering option to make the thing filterable? This would add a lot of complexity that would probably be more easily handled by the pluggability of the largo_opengraph function.

A quick fix for this issue in child themes is to define the function `largo_opengraph()` and have it output nothing, thereby replacing Largo's own [largo_opengraph](https://github.com/INN/largo/blob/c2ea897503ddada7f1687545d8af54915249b51f/inc/open-graph.php#L10-L95) output with no output.

**Comment by benlk** [comment]
https://github.com/INN/largo/pull/1589 removes Largo's largo_opengraph if Yoast is active, but that kicks the can down the line a while.

**Comment by benlk** [comment]
The "if yoast is active" check should just be a option in the theme options: 

> Disable Largo's built-in Twitter Card and Open Graph tags? If you're using an SEO plugin such as Yoast, you may need to check this box.

**Comment by benlk** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Add checkbox to disable Largo's SEO output, for plugin compatibility](https://trello.com/c/RbVyLpAz/42-add-checkbox-to-disable-largos-seo-output-for-plugin-compatibility)
